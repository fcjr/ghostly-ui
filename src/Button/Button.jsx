import React from 'react';
import './Button.scss';

const Button = ({
	autofocus = false,
	disabled = false,
	name = '',
	value = '',
	type = 'button',
	text,
	tier = 'primary',
}) => (
	<button
		autoFocus={autofocus}
		type={type}
		disabled={disabled}
		name={name}
		value={value}
		className={`Button__${tier}`}
	>
		{text}
	</button>
);

export default Button;
