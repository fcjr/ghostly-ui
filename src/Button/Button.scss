%buttonFormat {
	width: 184px;
	height: 50px;
	border: none;
	border-radius: 4px;
	font-family: 'Open Sans', sans-serif;
  	font-size: 16px;
  	font-weight: 300;
	color: #fff;
}

.Button__primary {
	@extend %buttonFormat;
	background-image: linear-gradient(105deg, #2b5993 3%, #00aef0 100%);
	&:hover {
		background-image: none;
		background-color: #2b5993;
	}
	&:disabled {
		background-image: none;
		background-color: rgba(43, 89, 147, 0.5);
	}
}

.Button__secondary {
	@extend %buttonFormat;
	background-color: #00aef0;
	&:hover {
		background-color: #0078ca;
	}
	&:disabled {
		opacity: .5;
		background-color: #7fbce5;
	}
}

.Button__alternate {
	@extend %buttonFormat;
	color: #4a4a4a;
	border: solid 2px #4a4a4a;
	background-color: #fff;
	&:hover, :active {
		background-color: #00aef0;
	}
	&:disabled {
		opacity: .5;
	}
}
