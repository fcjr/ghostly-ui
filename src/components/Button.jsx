import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { color, typography, spacing } from './shared/styles';

const StyledButton = styled.button`
	width: 184px;
	height: 50px;
	border: none;
	border-radius: ${spacing.borderRadius}px;
	font-family: ${typography.type.primary};
  	font-size: ${typography.size.p}px;
  	font-weight: ${typography.weight.regular};
	color: #FFFFFF;

	${(props) => props.appearance === 'primary' && css`
		background-image: linear-gradient(105deg, ${color.gradientleft} 3%, ${color.gradientright} 100%);
		&:hover {
			background-image: none;
			background-color: ${color.darkblue};
		}
		&:disabled {
			background-image: none;
			background-color: rgba(43, 89, 147, 0.5);
		}
	`}

	${(props) => props.appearance === 'secondary' && css`
		background-color: ${color.primary};
		&:hover {
			background-color: #0078ca;
		}
		&:disabled {
			opacity: .5;
			background-color: #7fbce5;
		}
	`}

	${(props) => props.appearance === 'alternate' && css`
		color: ${color.dark};
		border: solid 2px  ${color.dark};
		background-color: #FFFFFF;
		&:hover, :active {
			background-color: ${color.primary};
		}
		&:disabled {
			background-color: #FFFFFF;
			opacity: .5;
		}
`}
`;

export const Button = ({
	className,
	autofocus,
	disabled,
	name,
	value,
	type,
	text,
	appearance,
	...props
}) => (
	<StyledButton
		className={className}
		autoFocus={autofocus}
		type={type}
		disabled={disabled}
		name={name}
		value={value}
		appearance={appearance}
		{...props}
	>
		{text}
	</StyledButton>
);

Button.propTypes = {
	className: PropTypes.string,
	/**
	 * Whether the button should be autofocused
	 */
	autofocus: PropTypes.bool,
	/**
	 * Whether or not the button is in disabled state
	 */
	disabled: PropTypes.bool,
	/**
	 * The name attribute to be assigned to button
	 */
	name: PropTypes.string,
	/**
	 * The value attribute to be assigned to button
	 */
	value: PropTypes.string,
	/**
	 * The text shown inside the button
	 */
	text: PropTypes.string.isRequired,
	/**
	 * The type of button
	 */
	type: PropTypes.string,
	/**
	 * Determines button's color tier
	 */
	appearance: PropTypes.oneOf(['primary', 'secondary', 'alternate']).isRequired,
	/**
	 *  called on click
	*/
	onClick: PropTypes.func.isRequired
};

Button.defaultProps = {
	className: null,
	autofocus: false,
	disabled: false,
	name: '',
	value: '',
	type: 'button',
};
