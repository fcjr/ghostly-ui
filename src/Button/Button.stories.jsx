import React from 'react';

import Button from './Button';

export default {
	title: 'Inputs/Buttons/Button',
	component: Button,
};

const Template = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
	text: 'Primary',
	tier: 'primary',
};

export const Secondary = Template.bind({});
Secondary.args = {
	text: 'Secondary',
	tier: 'secondary',
};

export const Alternate = Template.bind({});
Alternate.args = {
	text: 'Alternate',
	tier: 'alternate',
};
