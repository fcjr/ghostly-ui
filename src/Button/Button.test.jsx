import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
	it('renders properly', () => {
		const tree = renderer
			.create(<Button text="Primary" tier="primary" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
