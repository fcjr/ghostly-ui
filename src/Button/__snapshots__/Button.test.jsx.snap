// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<button
  autoFocus={false}
  className="Button__primary"
  disabled={false}
  name=""
  type="button"
  value=""
>
  Primary
</button>
`;
